code/game/objects/structures/tables_racks.dm

```diff
@@ -208,6 +214,35 @@
 	material_flags = MATERIAL_ADD_PREFIX | MATERIAL_COLOR
 	buildstack = null //No buildstack, so generate from mat datums
 
+///Table on wheels
+/obj/structure/table/rolling
+	name = "Rolling table"
+	anchored = FALSE
+	smooth = SMOOTH_FALSE
+	canSmoothWith = list()
+	icon = 'icons/obj/smooth_structures/rollingtable.dmi'
+	icon_state = "rollingtable"
+	var/list/attached_items = list()
+
+/obj/structure/table/rolling/AfterPutItemOnTable(obj/item/I, mob/living/user)
+	. = ..()
+	attached_items += I
+	RegisterSignal(I, COMSIG_MOVABLE_MOVED, .proc/RemoveItemFromTable) //Listen for the pickup event, unregister on pick-up so we aren't moved
+
+/obj/structure/table/rolling/proc/RemoveItemFromTable()
+	attached_items -= source
+	UnregisterSignal(source, COMSIG_ITEM_PICKUP)
+
+/obj/structure/table/rolling/Move(atom/newloc, direct, glide_size_override)
+	var/turf/T = get_turf(src)
+	if(!..())
+		return FALSE
+	for(var/mob/M in T.contents)//Kidnap everyone on top
+		M.forceMove(newloc, direct)
+	for(var/x in attached_items)
+		var/atom/movable/AM = x
+		AM.Move(newloc, direct)

```

Comment:
This unlinks items. This should be called from Moved() instead, prevent items from getting unlinked AND use forceMove OR unlink if they fail to move.

Reply:
yeah my bad, I changed this while half-asleep but I still wanna do a bigger rework of this anyways